src/utils/ChildMapping.js

```diff
@@ -0,0 +1,87 @@
+import { Children } from 'react';
+
+
+/**
+ * Given `this.props.children`, return an object mapping key to child.
+ *
+ * @param {*} children `this.props.children`
+ * @return {object} Mapping of key to child
+ */
+export function getChildMapping(children) {
+  if (!children) {
+    return children;
+  }
+  let result = {};
+  Children.map(children, child => child)
+    .forEach((child) => {
+      result[child.key] = child;
+    });
```

Comment:
Does something like `Children.toArray(children)` help? I don't actually know this part of the code well.